py3status/modules/arch_updates.py

```diff
@@ -50,83 +39,88 @@ class Py3status:
     cache_timeout = 600
     format = None
     hide_if_zero = False
-    include_aur = False
+
+    class Meta:
+        deprecated = {"remove": [{"param": "include_aur", "msg": "obsolete"}]}
 
     def post_config_hook(self):
-        if not self.format:
-            if not self.include_aur:
-                self.format = FORMAT_PACMAN_ONLY
-            else:
-                self.format = FORMAT_PACMAN_AND_AUR
-        self.include_aur = self.py3.format_contains(self.format, "aur")
-        self.include_pacman = self.py3.format_contains(self.format, "pacman")
-        if self.py3.format_contains(self.format, "total"):
-            self.include_aur = True
-            self.include_pacman = True
-
-        if self.include_aur:
-            aurs = ["auracle", "yay", "cower"]
-            command = self.py3.check_commands(aurs)
-            if command:
-                self._check_aur_updates = getattr(
-                    self, "_check_aur_updates_{}".format(command)
-                )
+        pacman = self.py3.check_commands(["checkupdates"])
+        aur = self.py3.check_commands(["auracle", "yay", "cower"])
+
+        if self.format:
+            placeholders = self.py3.get_placeholders_list(self.format)
+            if "total" in placeholders:
+                pass
+            elif not pacman and not aur:
+                raise Exception(STRING_NOT_INSTALLED.format("pacman, aur"))
             else:
-                self.include_aur = False
-                self.py3.notify_user(STRING_NOT_INSTALLED.format(", ".join(aurs)))
-
-    def arch_updates(self):
-        pacman_updates = aur_updates = total = None
-
-        if self.include_pacman:
-            pacman_updates = self._check_pacman_updates()
-
-        if self.include_aur:
-            aur_updates = self._check_aur_updates()
+                for name, cmd in [("pacman", pacman), ("aur", aur)]:
+                    if name not in placeholders:
+                        cmd = None
+                    elif not cmd:
+                        raise Exception(STRING_NOT_INSTALLED.format(name))
+        elif pacman and aur:
+            self.format = "UPD: {pacman}/{aur}"
+        elif pacman:
+            self.format = "UPD: {pacman}"
+        elif aur:
+            self.format = "UPD: ?/{aur}"
```

Comment:
@maximbaz Since this is `auto`, do we want `UPD: {pacman}` or `UPD: {pacman}/?` I assume users prefer to install aur helpers too rather than using `{pacman}` alone. Also, if users really want to use `UPD: {pacman}`, they can specify in the `format`... so I have hard time seeing why users wouldn't want aur helpers too (other than fresh installations). Is it same for you?

Reply:
I'm not expert in formatting magic, what's the difference between `UPD: {pacman}` or `UPD: {pacman}/?`?

Reply:
No formatting magic. Show question mark suggesting users to install missing helpers. 

If `aur` is installed (and no pacman), I can do `UPD: {aur}` or `UPD: ?/{aur}`. I went with `UPD: ?/{aur}` to avoid misleading, but should I do the same thing for pacman (and no aur) `UPD: {pacman}/?` or use as-is `UPD: {pacman}`?

I think deprecating `hide_if_zero` is a strong no because of new formatter. I made changes to deprecate `include_aur`/detect auto... because I believe `include_aur = False` is a poor default config similar to no default buttons in `volume_status`.

Reply:
Aaaaaah, I got the question, sorry was confused at first 🙂 

It's nearly impossible to have pacman missing, pacman is almost what defines Arch, so feel free to keep `?/{aur}` :) As for the opposite case, no there's definitely a bunch of users who don't have any AUR helper and care only about official repos, so for sure `{pacman}` is preferable over `{pacman}/?`.

Reply:
Yup. Keep it as-is then. Better to default `format = '[\?if=total UPD: [{pacman}|\?show ?][/{aur}]]'` with `hide_if_zero = True` behavior.

Reply:
How about this? Which change do you like better?

* `UPD: 30/5`, `PAC: 30`, `AUR: 5` (latest change)
* `UPD: 30/5`, `UPD: 30`, `UPD: ?/5` (last change)
* `UPD: 30/5`, `UPD: 30`, `(empty)` (old change)

Reply:
Given that the last column is not possible to see, in practice we are choosing between `UPD: 30/5` `PAC: 30` and `UPD: 30/5` `UPD: 30`, I think the latter is slightly preferable just because if a user accidentally installs an AUR helper, module output will not drastically change (consistency!).

Reply:
>not possible to see

If you uninstalled `pacman-contrib` and kept `yay` or others, does it work okay? That package is part of `communtiy` too. I can see users having their favorite helper installed, but may install `pacman-contrib` later to fix `AUR` and/or `UPD ?/5`.

Reply:
oooh you are completely right! even though pacman itself is always present, the `checkupdates` script is part of contrib these days.

Reply:
This issue is trivial, but go with `last change` or `latest change`? Merge soon.

Reply:
still think `last change` is a tiny better :)

Reply:
Done.

Reply:
I'm waiting for your approval. No rush.

Reply:
will you give me a couple of days to prepare the alternative `updates` PR, to see if we can deprecate this module instead of changing it?

Reply:
Yes. Of course, but same issue as `updates` where we can't deprecate modules. Cheers.

Reply:
I thought we can't _remove_ modules, but can't we deprecate them just like we deprecate parameters? Basically all I want is to have a warning in log "you are using deprecated module X, switch to Y" - and let people still use the old module until we decide something about this in 4.0

Reply:
46c058ffae0d3e25e0b471b65b81fa3c3ef48967